services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/ValidationService.java

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.nbworker.services;
+
+import org.openkilda.messaging.info.rule.FlowEntry;
+import org.openkilda.messaging.info.rule.SwitchFlowEntries;
+import org.openkilda.messaging.info.switches.SyncRulesResponse;
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowSegment;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import com.google.common.collect.ImmutableSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class ValidationService {
+    private static final Logger logger = LoggerFactory.getLogger(ValidationService.class);
+    public static final Set<Long> SERVICE_RULES =
+            ImmutableSet.of(0x8000000000000001L, 0x8000000000000002L, 0x8000000000000003L, 0x8000000000000004L);
+    private SwitchFlowEntries request;
+    private RepositoryFactory repositoryFactory;
+    private SwitchId switchId;
+
+    public ValidationService(SwitchFlowEntries request, RepositoryFactory repositoryFactory) {
```

Comment:
If we exclude `SwitchFlowEntries` from here we can use one instance of `ValidationService` per bolt instance, not per tuple instance like now.